src/services/StreamService.ts

```diff
@@ -73,7 +73,7 @@ class StreamService {
         const templateRecord: any = Object.assign({}, record);
         Object.assign(templateRecord, {});
 
-        record.testTypes.forEach((testType: any, i: number, array: any[]) => {
+        record.testTypes?.forEach((testType: any, i: number, array: any[]) => {
```

Comment:
Not sure if there are already any there, but would be worth adding a quick test to verify this elvis prevents an error being thrown when a test result comes through without a testTypes array.